Fedoraware/Fedoraware-TF2/src/Features/Vars.h

```diff
@@ -2,920 +2,928 @@
-			inline CVar<int> SortMethod{ 1 }; //0 - FOV,		1 - Distance
-			inline CVar<bool> RespectFOV{ false };
-			inline CVar<bool> RequireBind{ true };
-			inline CVar<int> AimMethod{ 2 }; //0 - Normal,	1 - Smooth, 2 - Silent
-			inline CVar<int> SmoothingAmount{ 8 };
-			inline CVar<bool> RangeCheck{ false };
-			inline CVar<bool> PredictSwing{ false };
-			inline CVar<bool> WhipTeam{ false };
-		}
-	}
+inline std::vector<CVarBase*> g_Vars;
 
-	namespace NoSpread
-	{
-		inline CVar<bool> Hitscan{ false };
-		inline CVar<bool> Projectile{ true };
-		inline CVar<bool> Indicator{ false };
-		inline CVar<bool> ExtremePred{ false };
-		inline CVar<bool> CorrectPing{ false };
-		inline CVar<bool> UseAvgLatency{ false };
-	}
+template<class T>
+inline ConfigVar<T>::ConfigVar(T value, std::string name, bool visual)
+{
+	Value = value;
+	m_iType = typeid(T).hash_code();
+	m_sName = name;
+	g_Vars.push_back(this);
+	m_bVisual = visual;
+}
 
-	namespace Triggerbot
-	{
-		namespace Global
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<int> TriggerKey{ VK_XBUTTON2 };
-			inline CVar<int> IgnoreOptions{ 0b00000 }; //disguised, fakelagging players, taunting, friends, cloaked, invul
-		}
-
-		namespace Shoot
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<bool> TriggerPlayers{ false };
-			inline CVar<bool> TriggerBuildings{ false };
-			inline CVar<bool> HeadOnly{ false };
-			inline CVar<bool> WaitForHeadshot{ false };
-			inline CVar<bool> ScopeOnly{ false };
-			inline CVar<float> HeadScale{ 0.7f };
-		}
-
-		namespace Stab
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<bool> RageMode{ false };
-			inline CVar<bool> Silent{ false };
-			inline CVar<bool> Disguise{ false };
-			inline CVar<bool> IgnRazor{ false };
-			inline CVar<float> Range{ 0.9f };
-		}
-
-		namespace Detonate
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<int> DetonateTargets{ 0b00001 };
-			inline CVar<bool> Stickies{ false };
-			inline CVar<bool> Flares{ false };
-			inline CVar<float> RadiusScale{ 1.0f };
-		}
-
-		namespace Blast
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<bool> Rage{ false };
-			inline CVar<bool> Silent{ false };
-			inline CVar<bool> ExtinguishPlayers{ false };
-			inline CVar<bool> DisableOnAttack{ false };
-			inline CVar<int> Fov{ 55 };
-		}
-
-		namespace Uber
-		{
-			inline CVar<bool> Active{ false };
-			inline CVar<bool> OnlyFriends{ false };
-			inline CVar<bool> PopLocal{ false };
-			inline CVar<bool> AutoVacc{ false };
-			inline CVar<float> HealthLeft{ 35.0f };
-			inline CVar<bool> VoiceCommand{ false };
-			inline CVar<int> ReactFoV{25};
-			inline CVar<bool> BulletRes{ true }; inline CVar<bool> BlastRes{ true }; inline CVar<bool> FireRes{ true };
-			inline CVar<int> ReactClasses{0b000000000};	//	this is intuitive
-		}
-	}
+// These are dumb workarounds to make this compatible with the old system... Hope this works :)
+#define NAMESPACE_BEGIN(name) \
```

Comment:
Hmm but how could this be improved? Maybe we should switch to not using namespace at some point in the future (`Misc_Menu_Color` instead of `Misc::Menu::Color` for example)?